0021-zarf-values/README.md

```diff
@@ -0,0 +1,291 @@
+```
+**When** I deploy that package with a `zarf-config.yaml` like the below* or by specifying `-f values.yaml`:
+```yaml
+package:
+  deploy:
+    values:
+      - values.yaml
+```
+**And** I have a `values.yaml` like the below:
+```yaml
+component:
+  resources:
+    limits:
+      memory: 128Mi
+      cpu: 100m
+    requests:
+      memory: 64Mi
+      cpu: 100m
+
+other-component:
+  disabled: true
+```
+**Then** Zarf will apply the entire interface to the Helm chart override
+
+**Given** I have a Zarf Package with Go templating inside an action
+```yaml
+components:
+  - name: component
+    actions:
+      onDeploy:
+        before:
+          - cmd: "echo \"${{ .Values.component.resources.limits.memory }}\""
+```
+**When** I deploy that package with a `zarf-config.yaml` like the below* or by specifying `-f values.yaml`:
+```yaml
+package:
+  deploy:
+    values:
+      - values.yaml
+```
+**And** I have a `values.yaml` like the below:
+```yaml
+component:
+  resources:
+    limits:
+      memory: 128Mi
+      cpu: 100m
+    requests:
+      memory: 64Mi
+      cpu: 100m
+
+other-component:
+  disabled: true
+```
+**Then** Zarf will template the action and output `128Mi`
+
+**Given** I have a Zarf Package with a setValues action
+```yaml
+components:
+  - name: component
+    actions:
+      onDeploy:
+        before:
+          - cmd: "echo '{ \"memory\": \"256Mi\", \"cpu\": \"200m\" }'"
+            setValues:
+              - path: component.resources.limits
+                type: json
+                
+```
+**When** I deploy that package with a `zarf-config.yaml` like the below* or by specifying `-f values.yaml`:
+```yaml
+package:
+  deploy:
+    values:
+      - values.yaml
+```
+**And** I have a `values.yaml` like the below:
+```yaml
+component:
+  resources:
+    limits:
+      memory: 128Mi
+      cpu: 100m
+    requests:
+      memory: 64Mi
+      cpu: 100m
+
+other-component:
+  disabled: true
+```
+**Then** Zarf will initially use the values provided for any components before the one with `setValues`
+**And Then** apply the values provided in the `setValues` action for any following components based on the actions lifecycle.
+
+> [!NOTE]
+> *This would apply to all `zarf-config` formats not just YAML
+
+### Risks and Mitigations
+
+This will introduce a wholly new way to input values into Zarf that will live alongside the existing Variables, Constants and Templates for now.  Because of this, the feature will need to be clearly disambiguated from Variables/Constants/Templates in documentation and while this feature should not introduce many breaking changes being implemented alongside the existing featureset, the feature to map Zarf Variables to Helm Values should be deprecated and removed in favor of the new Zarf Values mapping to assist with disambiguation.  If the feature gains traction and is accepted by the community, a deprecation plan for the original Zarf Variables/Constants/Templates should be created.
```

Comment:
Likely we will not break `.components.charts.variables` ondeploy for existing packages, but will not allow new packages to be created using this paradigm when values are released.

Reply:
added a note to this effect.